src/download.tsx

```diff
@@ -143,11 +144,11 @@ class Download extends KalturaPlayer.core.BasePlugin {
     await this.ready;
 
     this.downloadPluginManager.setShowOverlay(false, false);
-    const downloadMetadata = await this.downloadPluginManager.getDownloadMetadata(true);
+    this.downloadMetadata = await this.downloadPluginManager.getDownloadMetadata(true);
 
-    if (this.shouldInjectUI(downloadMetadata)) {
+    if (this.isPluginSupported()) {
       this.logger.debug('Download is supported for current entry');
-      this.injectOverlayComponents(downloadMetadata);
+      this.injectOverlayComponents(this.downloadMetadata);
```

Comment:
if this is a class member now, why also pass it as argument

Reply:
Pure function are always better architecture and easy to test, you can read about functional programming 


functional programming -  architectural style, where functions receive all necessary data through parameters rather than relying on external state, is often referred to as **functional programming**. In JavaScript, this can be part of a broader design pattern sometimes called **functional programming paradigm** or simply **pure functions**.

### Benefits of Using Pure Functions and Functional Programming:

1. **Predictability**:
   - Pure functions always produce the same output for the same input, making the behavior of the code predictable and consistent.

2. **Testability**:
   - Pure functions are easier to test because they don't depend on external state. You can test them in isolation by providing input and verifying the output.

3. **Debugging**:
   - Since pure functions do not cause side effects, debugging is simplified. If something goes wrong, you know it's due to the function's logic, not some hidden state change.

4. **Reusability**:
   - Pure functions are more reusable because they don’t rely on the context or state. They can be used in different parts of the application without modification.

5. **Maintainability**:
   - Code is easier to maintain because functions are independent and have well-defined interfaces. Changes in one part of the system are less likely to affect other parts.

6. **Concurrency**:
   - Pure functions are inherently thread-safe and can be executed concurrently without worrying about race conditions or shared state issues.

7. **Composability**:
   - Pure functions can be composed together to build more complex operations. This leads to more modular and flexible code.

8. **Immutability**:
   - Functional programming often emphasizes immutability, which helps prevent unexpected changes in state, making the code more robust.

9. **Readability**:
   - Functional code is often more concise and easier to read, as it focuses on what to do rather than how to do it.

10. **Reduced Bugs**:
    - By avoiding side effects and mutable state, the potential for bugs related to state changes is minimized.

This approach can significantly improve the quality and reliability of your code, especially in complex applications.